Modules/posixmodule.c

```diff
@@ -5088,384 +5088,392 @@ os__path_splitroot_impl(PyObject *module, path_t *path)
 }
 
 
-/*[clinic input]
-os._path_isdir
-
-    s: 'O'
+#define PY_IFREG  1 // Regular file
+#define PY_IFDIR  2 // Directory
+#define PY_IFLNK  4 // Symlink
+#define PY_IFMNT  8 // Mount Point (junction)
+#define PY_IFLRP 16 // Link Reparse Point (name-surrogate, symlink, junction)
+#define PY_IFRRP 32 // Regular Reparse Point
 
-Return true if the pathname refers to an existing directory.
-
-[clinic start generated code]*/
+static inline BOOL
+_testReparseTag(DWORD reparseTag, int testedType)
```

Comment:
This could be used more generally if `attributes` is added as a parameter. The stat fallback can switch to using `st_file_attributes` instead of `st_mode`.